clash-lib/src/Clash/Core/Term.hs

```diff
@@ -126,6 +139,57 @@ data Pat
 
 type Alt = (Pat,Term)
 
+-- | Get the list of term-binders out of a DataType pattern
+patIds :: Pat -> ([TyVar],[Id])
+patIds (DataPat _  tvs ids) = (tvs,ids)
+patIds _                    = ([],[])
+
+patVars :: Pat -> [Var a]
+patVars (DataPat _ tvs ids) = coerce tvs ++ coerce ids
+patVars _ = []
+
+-- | Abstract a term over a list of term and type variables
+mkAbstraction :: Term
+              -> [Either Id TyVar]
+              -> Term
```

Comment:
If you feel like it: reformat these functions to adhere to the styleguide (i.e., type signature on a single line OR the `::` on the next line.

Reply:
I always feel like making the code prettier, I just didn't want to make all the diffs confusing
If this gets approved, how about I squash the two pruning commits and then do the stylistic changes on the squashed commit?

Reply:
Works for me, no need to squash IMO.

Reply:
Fair, in that case I'll leave a style fix commit at the end of the PR so the history doesn't become more tedious to read